packages/server-wallet/src/chain-service/__chain-test__/chain-service.test.ts

```diff
@@ -308,6 +320,26 @@ describe('registerChannel', () => {
     );
   });
 
+  it('Receives 0 initial holding when holdings are not confirmed', async () => {
+    const channelId = randomChannelId();
+    const expected = [0, 5, 8, 10];
+    waitForChannelFunding(0, 5, channelId);
+    waitForChannelFunding(5, 3, channelId);
+    const p = new Promise(resolve => {
+      chainService.registerChannel(channelId, [ethAssetHolderAddress], {
+        ...defaultNoopListeners,
+        holdingUpdated: arg => {
+          expect(arg.amount).toEqual(BN.from(expected[0]));
+          expected.shift();
+          if (!expected.length) resolve(true);
+        },
+      });
+    });
+    fundChannel(8, 2, channelId, ethAssetHolderAddress);
+    mineBlocksSlow(20);
```

Comment:
Does this add 10 seconds to the testsuite run?

Reply:
No - we don't wait for this, test case will finish without waiting for all async operations to finish.

Reply:
Got it! Our approach in the past has been to mine the minimum number of blocks for a test to pass. This has the benefit of tests not interfering with each other. An example is the following. Let's say we have a test to verify that a callback is invoked 5 blocks after a transaction. We would set up the test in the following way:
1. Mine the transaction.
2. Mine 2 blocks.
3. Verify that the callback has not been invoked.
4. Mine 3 more blocks.
5. Verify that the callback has been invoked.

Mining extra blocks in a test has the unfortunate side-effect of not being able to write tests using the approach above.

Reply:
I'll  modify it to await the mining process to finish before moving to the next case. The exact amount of time to wait will be hard to find. I'll find a value that is safe.

Reply:
By "the exact amount of time" do you mean relying on timeouts for testing? If so, would mind describing a scenario you are thinking of?

Reply:
that's from your original question, we don't want to wait for 10s unless necessary.

I've added await to make sure mineBlockSlow finishes before test case exists, but I suspect that is not the only reason that test cases are affecting each others - chain seems to move faster when multiple test cases run together.

Reply:
If we can assume that each test case waits for all of the blocks the test case mines, the hope is a test case does not affect other test cases. The chain state is not reset between test cases, but each test uses random channel ids.

That being said, perhaps there are other ways that tests interfere with each other. If you notice behavior like that, it would be great to hear about it!

Reply:
FYI: we use the `--runInBand` jest flag which forces tests to execute sequentially.